src/core/vector/qgsvectorlayer.cpp

```diff
@@ -1373,6 +1373,11 @@ bool QgsVectorLayer::addFeature( QgsFeature &feature, Flags )
       success = mJoinBuffer->addFeature( feature );
   }
 
+  if ( success && !mDefaultValueOnUpdateFields.isEmpty() )
+  {
+    updateDefaultValues( feature.id() );
```

Comment:
@signedav , one thing I'm a tiny bit worried about here is the impact it'll have one default values that might rely on attribute form context and expression scopes (e.g. @current_parent). 

Could we do that on the attribute form side of things?

Reply:
Thanks.

The reason why this is not done on attribute form side is, that (1) it should consider fields that are not added to the form via drag'n'drop as well and (2) to be aligned with the behavior on editing an existent feature. 

I understand the "weakness" that values could be changed twice (like a timestamp that is stored some seconds later than shown in the form before) but I don't understand what you mean with the @current_parent. Could you clarify the use case?